interview-mr/src/test/java/com/sovrn/interview/mr/AverageScoreTest.java

```diff
@@ -0,0 +1,93 @@
+package com.sovrn.interview.mr;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.junit.Assert.assertEquals;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+
+import com.sovrn.interview.mr.AverageScore;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.Before;
+import org.junit.Test;
+
+public class AverageScoreTest {
+	private final String outputPath = "junit-output";
+	AverageScore score;
+	
+	@Before
+	public void clean() {
+		try {
+			FileUtils.deleteDirectory(new File(outputPath));
+		} catch (IOException e) {
+			// ignore
+		}
+	}
+	
+	private Map<String, Float> getOutput() {
+		Scanner scanner = null;
+		Map<String, Float> results = new HashMap<String, Float>();
+		
+		try {
+			scanner = new Scanner(new File(outputPath+"/part-r-00000"));
+			while(scanner.hasNext()) {
+				String values[] = scanner.nextLine().split("\t");
+				if(values.length < 2)
+					fail("Output contained invalid line");
+				results.put(values[0], Float.parseFloat(values[1]));
+			}
+		} catch (FileNotFoundException e) {
+			fail("Output file not found: "+e);
+		} catch (NumberFormatException e) {
+			fail("Output file contained data without a numerical value: "+e);
+		} finally {
+			if(scanner != null)
+			    scanner.close();
+		}
+		return results;
+	}
+	
+	@Test
+	public void mapReduceBasicTest() {
+		try {
+			int retVal = AverageScore.mapReduce("src/main/data/data.tsv", outputPath);
+			assertEquals(0,retVal);
+			
+			// Verify basic output format and number of entries
+			assertEquals(189, getOutput().size());
+		} catch (Exception e) {
+			fail("Exceptions should not be thrown, but threw "+e);
+		}
+	}
+	
+	@Test
+	public void mapReduceSpecificTest() {
```

Comment:
Kind of an 80/20 effort here. Tests all the basic edge cases, but doesn't get into many sequences/combinations of possible errors. Further testing would see diminishing returns.

A reasonable next addition would be a very large scale average computation, but that's a big piece of work to bite off.